### PR TITLE
Funnel state -> states-dev

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

**Review by hillelv — DISMISSED** [review]

**Review by Nitay4 — DISMISSED** [review]
noicee

File: src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	private final static SparkLimitSwitch.Type FORWARD_LIMIT_SWITCH_TYPE = SparkLimitSwitch.Type.kNormallyOpen;
+
+
+	public static FunnelStuff generateFunnelStuff(String logPath) {
+		SparkMaxWrapper sparkMAXWrapper = new SparkMaxWrapper(IDs.CANSparkMAXIDs.FUNNEL);
+
+		SysIdRoutine.Config config = new SysIdRoutine.Config();
+		sparkMAXWrapper.setIdleMode(CANSparkBase.IdleMode.kCoast);
+
+		sparkMAXWrapper.setSmartCurrentLimit(30);
```
Comment (Yotam7) [inline src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java:31]: magic number

**Review by Yotam7 — DISMISSED** [review]

**Review by YHerm — DISMISSED** [review]

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -17,9 +20,12 @@ public class Robot {
 
 	public static final RobotType ROBOT_TYPE = RobotType.determineRobotType();
 
+	private final Funnel funnel;
+
 	private final Superstructure superstructure;
 
 	public Robot() {
+		this.funnel = new Funnel(FunnelFactory.create(FunnelConstants.LOG_PATH));
 		this.superstructure = new Superstructure(this);
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/Robot.java:41]: Add a space

File: src/main/java/frc/robot/subsystems/funnel/FunnelState.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package frc.robot.subsystems.funnel;
+
+public enum FunnelState {
+
+	NOTE_TO_SHOOTER(0.1),
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/funnel/FunnelState.java]: No outtake? And orginize this

File: src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import frc.robot.hardware.motor.sparkmax.SparkMaxWrapper;
+import frc.robot.hardware.signal.supplied.SuppliedDoubleSignal;
+import frc.robot.subsystems.funnel.FunnelConstants;
+import frc.robot.subsystems.funnel.FunnelStuff;
+
+import java.util.function.BooleanSupplier;
+
+public class RealFunnelConstants {
+
+	private final static double DEBOUNCE_TIME_SECONDS = 0.05;
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java:19]: Spaces

File: src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+	public static FunnelStuff generateFunnelStuff(String logPath) {
+		SparkMaxWrapper sparkMAXWrapper = new SparkMaxWrapper(IDs.CANSparkMAXIDs.FUNNEL);
+		configMotor(sparkMAXWrapper);
+		BrushlessSparkMAXMotor motor = new BrushlessSparkMAXMotor(logPath, sparkMAXWrapper, new SysIdRoutine.Config());
+
+		SuppliedDoubleSignal voltageSignal = new SuppliedDoubleSignal("voltage", sparkMAXWrapper::getVoltage);
+
+		BooleanSupplier isShooterBeamBroken = () -> sparkMAXWrapper.getReverseLimitSwitch(REVERSE_LIMIT_SWITCH_TYPE).isPressed();
+		sparkMAXWrapper.getReverseLimitSwitch(REVERSE_LIMIT_SWITCH_TYPE).enableLimitSwitch(false);
```
Comment (bossyshrimp68) [inline src/main/java/frc/robot/subsystems/funnel/factory/RealFunnelConstants.java]: Const

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]
This isnt updated to funnel

File: src/main/java/frc/robot/subsystems/funnel/FunnelStateHandler.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	private final Funnel funnel;
+
+	public FunnelStateHandler(Robot robot) {
+		this.funnel = robot.getFunnel();
+	}
+
+	public Command setState(FunnelState state) {
+		if (state == FunnelState.STOP) {
+			return funnel.getCommandsBuilder().stop();
+		}
```
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/funnel/FunnelStateHandler.java:17]: you don't really need this

File: src/main/java/frc/robot/subsystems/funnel/FunnelStateHandler.java
Comment (hillelv) [inline src/main/java/frc/robot/subsystems/funnel/FunnelStateHandler.java:1]: consider moving this to the commands builder

Comment (hillelv) [inline src/main/java/frc/robot/subsystems/funnel/FunnelStateHandler.java:1]: you don't have stopping conditions, the funnel will run until a different state is set, not regarding the note or the beambreakers

**Pull request closed without merging**